### PR TITLE
Sitepermission microphone

### PR DESCRIPTION
**Pull request by Amejia481**
[Demo 🌮 ](https://drive.google.com/file/d/14EJu0bTlAqCibeEaHCoC2e_tKPUiGDK3/view?usp=sharing)

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt [hunk 1]
```diff
@@ -188,6 +206,28 @@ class SitePermissionsFeature(
...
+        context: Context,
+        uriString: String,
+        buttons: List<DoorhangerPrompt.Button>
+    ): DoorhangerPrompt {
+
+        val title = context.getString(R.string.mozac_feature_sitepermissions_microfone_title, uriString)
+        val optionTitle = context.getString(R.string.mozac_feature_sitepermissions_option_microphone_one)
+        val drawable = ContextCompat.getDrawable(context, R.drawable.mozac_ic_microphone)
+
+        val option1 = RadioButton(optionTitle).apply { checked = true }
```
Comment (pocmo) [inline components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt]: "option1" ?

Comment (Amejia481) [reply components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt]: Naming is hard XD. I will change it :)

In the mock-ups they have a fine-grained microphone list, but right now GV is only providing us one (the default microphone) I guess in the future they are going to provide that.  
image

File: components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt [hunk 2]
```diff
@@ -188,6 +206,28 @@ class SitePermissionsFeature(
...
+        context: Context,
+        uriString: String,
+        buttons: List<DoorhangerPrompt.Button>
+    ): DoorhangerPrompt {
+
+        val title = context.getString(R.string.mozac_feature_sitepermissions_microfone_title, uriString)
+        val optionTitle = context.getString(R.string.mozac_feature_sitepermissions_option_microphone_one)
+        val drawable = ContextCompat.getDrawable(context, R.drawable.mozac_ic_microphone)
+
+        val microphoneRadioButton = RadioButton(optionTitle).apply { checked = true }
```
Comment (pocmo) [inline components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt]: nit: No need to apply "checked" here. The first radio button in a group is always checked.

Comment (Amejia481) [reply components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt]: Awesome!

File: components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt (hunk 1 above)
Comment (pocmo) [reply components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt]: Is this also the case if there are external microphones available?

Comment (Amejia481) [reply components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt]: I'm not quite sure, I'm just guessing from what I saw in the [mock-ups](https://github.com/mozilla-mobile/fenix/issues/332). They didn't include a mock-up with external microphones apart from the headset one.

**Pull request merged**